tests/lib/tools/tests.session

```diff
@@ -316,76 +262,37 @@ BEGIN {
-			fi
-			;;
-	esac
-
-	# NOTE: This shellcheck directive is for the $selinux_context_arg expansion below.
-	# shellcheck disable=SC2086
-	busctl \
-		--allow-interactive-authorization=no --quiet \
-		--system \
-		-- \
-		call \
-		 org.freedesktop.systemd1 \
-		/org/freedesktop/systemd1 \
-		 org.freedesktop.systemd1.Manager \
-		StartTransientUnit "ssa(sv)a(sa(sv))" \
-		"$unit_name" fail $(( 4 + $(test -n "$selinux_context_arg" && echo 1 || echo 0))) \
-			Description s "tests.session running $* as $user" \
-			Type s oneshot \
-			Environment as 1 TERM=xterm-256color \
-			$selinux_context_arg \
-			ExecStart "a(sasb)" 1 \
-				"$(command -v runuser)" 6 "$(command -v runuser)" -l "$user" - -c "SNAPD_DEBUG=$SNAPD_DEBUG SNAP_CONFINE_DEBUG=$SNAP_CONFINE_DEBUG exec $tmp_dir/exec <$tmp_dir/stdin.pipe >$tmp_dir/stdout.pipe 2>$tmp_dir/stderr.pipe" false \
-		0
-	# This is done so that we can configure file redirects. Once Ubuntu 16.04 is no
-	# longer supported we can use Standard{Input,Output,Error}=file:path property
-	# of systemd, or perhaps even StandardInputFileDescriptor and pass a file
-	# descriptor to the FIFOs we open in the script above.
-
-	# Wait for the service to terminate. The trigger is a successful read
-	# from the result FIFO. In case we are signaled in one of the familiar
-	# ways, kill the started service with the same signal. The reading occurs
-	# in a loop as the signal will first interrupt the read process, which will
-	# fail and return nothing.
-	trap 'systemctl kill --signal=INT $unit_name' INT
-	trap 'systemctl kill --signal=TERM $unit_name' TERM
-	trap 'systemctl kill --signal=QUIT $unit_name' QUIT
-	result=""
-	for sig in INT TERM QUIT; do
-		result=$(cat "$tmp_dir/result.pipe" 2>/dev/null) && break
-		trap - "$sig"
-	done
-	trap - INT TERM QUIT
-
-	# Kill dbus-monitor that otherwise runs until it notices the pipe is no longer
-	# connected, which happens after a longer while. Redirect stderr to /dev/null
-	# to avoid upsetting tests which are sensitive to stderr, e.g. tests/main/document-portal-activation
-	kill $dbus_monitor_pid 2>/dev/null || true
-	wait $dbus_monitor_pid 2>/dev/null || true
-	wait $awk_pid
-
-	wait "$cat_stdin_pid"
-	wait "$cat_stdout_pid"
-	wait "$cat_stderr_pid"
+	#selinux_context_arg=
+	#case $SPREAD_SYSTEM in
+    #	fedora-*|centos-*)
+	#		if [ "$(systemctl --version | head -n 1 | awk '{ print $2 }')" -gt 219 ]; then
+	#			selinux_context_arg="SELinuxContext s unconfined_u:unconfined_r:unconfined_t:s0"
+	#		fi
+	#		;;
+	#esac
+
+	systemd-run \
+		--description "tests.session running $* as $user" \
+		--quiet \
+		--collect \
+		--service-type oneshot \
+		--setenv TERM=xterm-256color \
+		--property "StandardInput=file:$tmp_dir/stdin.pipe" \
+		--property "StandardOutput=file:$tmp_dir/stdout.pipe" \
+		--property "StandardError=file:$tmp_dir/stderr.pipe" \
+		"$(command -v runuser)" -l "$user" - -c "SNAPD_DEBUG=$SNAPD_DEBUG SNAP_CONFINE_DEBUG=$SNAP_CONFINE_DEBUG exec $tmp_dir/exec"
```

Comment:
What about `--property "User=$user"`?
And `--property Environment=SNAPD_DEBUG=$SNAPD_DEBUG SNAP_CONFINE_DEBUG=$SNAP_CONFINE_DEBUG`?

Reply:
I wonder if `--uid=` would be better then.

Reply:
Which test failed with `User=`? I am interested in understanding why it fails.